File: fib.java
import java.io.*;
import java.util.*;
import java.util.concurrent.TimeUnit;

public class fib{
	public static void main(String[] args){
		System.out.print("Enter n: ");
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		
		System.out.println("Recursive:");
		long startTime = System.nanoTime();
		System.out.println(rec_fib(n));
		long endTime = System.nanoTime();
        	long timeElapsed = endTime - startTime;
        
        	System.out.println("Recursive Time required (ns):"+timeElapsed);
        
		
		System.out.println("Iterative:");
		long itestartTime = System.nanoTime();
		System.out.println(ite_fib(n));
		long iteendTime = System.nanoTime();
        	long itetimeElapsed = iteendTime - itestartTime;
        	
        	System.out.println("Iterative Time required (ns):"+itetimeElapsed);
		
	}
	public static int rec_fib(int n){
		if (n<=1){
			return n;
		}
		return (rec_fib(n-1)+rec_fib(n-2));
	}
	
	public static int ite_fib(int n){
		int a = 0;
		int b = 1;
		int c;
		for(int i=0;i<n;i++){
			c = a+b; 
			a = b;
			b = c;
		}
		return a;
	}
}

File: DAA/Assignment 1/Fib.py
import time

def rec_fib(n):
	if n<=1:
		return n
	return (rec_fib(n-1) + rec_fib(n-2))

def ite_fib(n):
	a = 0
	b = 1
	for i in range(n):
		a,b = b,a+b
	return a

n = int(input("Enter n: "))

start = time.time()
print(rec_fib(n))
end = time.time()
time_n = end - start

print("Recursive Time complexity: ",time_n)

start = time.time()
print(ite_fib(n))
end = time.time()
time_n = end - start

print("Iterative Time complexity: ",time_n)
print("Iterative Space complexity: ",4*5,"bytes")
